machine/machine_core/ssh_connection.py

```diff
@@ -314,8 +314,7 @@ def execute(self, command, input=None, environment={},
 
         with timeoutlib.Timeout(seconds=timeout, error_message="Timed out on '%s'" % command, machine=self):
             res = subprocess.run(command_line,
-                                 stdin=None if input is not None else subprocess.DEVNULL,
-                                 input=input.encode("UTF-8") if input else None,
+                                 input=input.encode("UTF-8") if input else b'',
```

Comment:
Why change it to `b''` the default is `None`, isn't our main issue using `stdin` and `input` together. Input also defaults to `None` so that sounds good.

Reply:
Because I'm not sure if `None` means "pipe" "`/dev/null`" or "empty string".  I'm 100% sure what `b''` means.

Reply:
This fixes the problem, as shown by cockpit-project/cockpit#17650

Reply:
Good question what `None` does the docs say:

```
The input argument is passed to [Popen.communicate()](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate) and thus to the subprocess’s stdin. If used it must be a byte sequence, or a string if encoding or errors is specified or text is true. When used, the internal [Popen](https://docs.python.org/3/library/subprocess.html#subprocess.Popen) object is automatically created with stdin=PIPE, and the stdin argument may not be used as well.
```

Anyway this works so let's merge it.